engine/src/main/kotlin/world/gregs/voidps/engine/entity/character/player/skill/level/Interpolation.kt

```diff
@@ -11,7 +11,12 @@ object Interpolation {
         return (y1 * (x2 - x) + y2 * (x - x1)) / (x2 - x1)
     }
 
+    fun interpolate(start: Double, end: Double, fraction: Double): Double {
+        return start + fraction * (end - start)
+    }
```

Comment:
Double precision is going to cause some really weird xp drops, like 15.4999999, it would be good to round all results to 1 decimal place.